tests/test_main.py

```diff
@@ -0,0 +1,371 @@
+import json
+
+import pytest
+from asgiref.sync import sync_to_async
+from django.contrib.contenttypes.models import ContentType
+from django.core import management
+from django.urls import reverse
+from django.utils.version import get_version
+from pytest_django.asserts import assertInHTML
+
+from easyaudit.middleware.easyaudit import clear_request, set_current_user
+from easyaudit.models import CRUDEvent, RequestEvent
+from tests.test_app.models import (
+    BigIntForeignKeyModel,
+    BigIntM2MModel,
+    BigIntModel,
+    ForeignKeyModel,
+    M2MModel,
+    Model,
+    UUIDForeignKeyModel,
+    UUIDM2MModel,
+    UUIDModel,
+)
+
+
+@pytest.mark.django_db
+def test_no_migrations(capsys: pytest.CaptureFixture):
+    management.call_command("makemigrations", dry_run=True)
+
+    captured = capsys.readouterr().out
+    assert "No changes detected" in captured
+
+
+def test_no_issues(capsys: pytest.CaptureFixture):
+    management.call_command("check", fail_level="WARNING")
+
+    captured: str = capsys.readouterr().out
+    assert "System check identified no issues" in captured
+
+
+@pytest.mark.parametrize(
+    "model",
+    [
+        BigIntForeignKeyModel,
+        BigIntM2MModel,
+        BigIntModel,
+        ForeignKeyModel,
+        M2MModel,
+        Model,
+        UUIDForeignKeyModel,
+        UUIDM2MModel,
+        UUIDModel,
+    ],
+)
+class TestDjangoCompat:
+    def test_model_state(self, model):
+        """Ensures models have the internal `_state` object."""
+        model_instances = model()
+        assert hasattr(model_instances, "_state")
+
+
+@pytest.mark.django_db
+class TestAuditModels:
+    @pytest.fixture
+    def model(self):
```

Comment:
The previous class attributes were moved to fixtures. This is the pytest way, and adds modularity since one can now add logic to these functions on subclasses if necessary.